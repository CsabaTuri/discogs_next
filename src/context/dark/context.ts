import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"

let initialState: any = { isDark: true, dispatch: () => { } }

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useDarkContext = () => useContext(Context);

export default Context

