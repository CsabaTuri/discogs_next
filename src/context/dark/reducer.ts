export function reducer(state:any, action:any) {
  switch (action.type) {
    case "isDark":
      return  {isDark: !state.isDark }
    default:
      return state
  }
}
