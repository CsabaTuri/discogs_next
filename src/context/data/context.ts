import { createContext ,useContext} from "react"

let initialState:any = {  data: {} ,dispatch: () => {}  };

 const Context = createContext(initialState)

 export const useDataContext = () => useContext(Context);

 export default Context
