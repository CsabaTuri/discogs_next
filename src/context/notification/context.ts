import { createContext, useContext } from "react";

let initialState:any = { state: { text: "", type: "success" },dispatch: () => {}  };

const Context = createContext(initialState);

export const useNotificationContext = () => useContext(Context);

export default Context;
