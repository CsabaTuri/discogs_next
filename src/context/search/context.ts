import { createContext ,useContext} from "react"

let initialState:any = { search:{search: {}},dispatch: () => {} }

 const Context = createContext(initialState)

 export const useSearchContext = () => useContext(Context);

 export default Context
