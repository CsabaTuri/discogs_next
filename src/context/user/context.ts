import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"



let initialState: any = { user: {}, dispatch: () => { } };

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useUserContext = () => useContext(Context);

export default Context
