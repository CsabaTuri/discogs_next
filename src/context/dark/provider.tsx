import React, { useCallback, useReducer } from "react";
import { useLocalStorage } from "react-use";
import { reducer } from "./reducer";
import Context from "./context";

let initialState: any = { isDark: true, dispatch: () => { } };

const usePersistReducer = () => {
  // grab saved value from `localStorage` and
  // a function to update it. if
  // no value is retrieved, use `INITIAL_STATE`
  const [savedState, saveState] = useLocalStorage<any>("isDark", initialState);

  // wrap `reducer` with a memoized function that
  // syncs the `newState` to `localStorage` before
  // returning `newState`. memoizing is important!
  const reducerLocalStorage = useCallback(
    (state, action) => {
      const newState = reducer(state, action);

      saveState(newState);

      return newState;
    },
    [saveState]
  );

  // use wrapped reducer and the saved value from
  // `localStorage` as params to `useReducer`.
  // this will return `[state, dispatch]`
  return useReducer(reducerLocalStorage, savedState);
};

export const Provider = ({ children }: any) => {
  const [state, dispatch] = usePersistReducer();
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
};
