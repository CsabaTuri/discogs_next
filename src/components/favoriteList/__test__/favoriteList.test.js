import React from "react";
import { render, screen, mount } from "@testing-library/react";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import FavoriteList from "../favoriteList";
import * as DarkContext from "../../../context/dark/context";
import "jest-styled-components";

Enzyme.configure({ adapter: new Adapter() });
const testFavoriteListData = {
  title: "Release",
  data: [
    {
      __typename: "FavReleases",
      release_name: "Velvet Acid Christ - Fun With Knives",
      release_id: "10521",
    },
    {
      __typename: "FavReleases",
      release_name: "X-102 - Titan EP",
      release_id: "1393615",
    },
    {
      __typename: "FavReleases",
      release_name: "Caetano* E Chico* - Juntos E Ao Vivo",
      release_id: "2664677",
    },
  ],
  mutation: {
    kind: "Document",
    definitions: [
      {
        kind: "OperationDefinition",
        operation: "mutation",
        name: { kind: "Name", value: "removeReleaseFromFavorites" },
        variableDefinitions: [
          {
            kind: "VariableDefinition",
            variable: { kind: "Variable", name: { kind: "Name", value: "id" } },
            type: {
              kind: "NonNullType",
              type: { kind: "NamedType", name: { kind: "Name", value: "ID" } },
            },
            directives: [],
          },
        ],
        directives: [],
        selectionSet: {
          kind: "SelectionSet",
          selections: [
            {
              kind: "Field",
              name: { kind: "Name", value: "remove_release_from_favorites" },
              arguments: [
                {
                  kind: "Argument",
                  name: { kind: "Name", value: "release_id" },
                  value: {
                    kind: "Variable",
                    name: { kind: "Name", value: "id" },
                  },
                },
              ],
              directives: [],
            },
          ],
        },
      },
    ],
    loc: { start: 0, end: 106 },
  },
  type: "release",
};
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "/",
      query: "",
      asPath: "/",
    };
  },
}));
jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));
describe("FavoriteList", () => {
  test("snapshot test", async () => {
    await new Promise((resolve) => {
      window.setTimeout(resolve, 0);
    });

    const contextValues = { state: { isDark: false } };
    jest
      .spyOn(DarkContext, "useDarkContext")
      .mockImplementation(() => contextValues);
    const wrapper = Enzyme.mount(
      <FavoriteList
        data={testFavoriteListData.data}
        mutation={testFavoriteListData.mutation}
        type={testFavoriteListData.type}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
