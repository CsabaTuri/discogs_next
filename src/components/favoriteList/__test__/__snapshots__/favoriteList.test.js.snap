// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList snapshot test 1`] = `
<FavoriteList
  data={
    Array [
      Object {
        "__typename": "FavReleases",
        "release_id": "10521",
        "release_name": "Velvet Acid Christ - Fun With Knives",
      },
      Object {
        "__typename": "FavReleases",
        "release_id": "1393615",
        "release_name": "X-102 - Titan EP",
      },
      Object {
        "__typename": "FavReleases",
        "release_id": "2664677",
        "release_name": "Caetano* E Chico* - Juntos E Ao Vivo",
      },
    ]
  }
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "removeReleaseFromFavorites",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "release_id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "remove_release_from_favorites",
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 106,
        "start": 0,
      },
    }
  }
  type="release"
>
  <Link
    href="/release/10521"
  >
    <a
      href="/release/10521"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Velvet Acid Christ - Fun With Knives
    </a>
  </Link>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FaTrash>
        <IconBase
          attr={
            Object {
              "viewBox": "0 0 448 512",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              key="0"
            />
          </svg>
        </IconBase>
      </FaTrash>
    </button>
  </Button>
  <br />
  <Link
    href="/release/1393615"
  >
    <a
      href="/release/1393615"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      X-102 - Titan EP
    </a>
  </Link>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FaTrash>
        <IconBase
          attr={
            Object {
              "viewBox": "0 0 448 512",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              key="0"
            />
          </svg>
        </IconBase>
      </FaTrash>
    </button>
  </Button>
  <br />
  <Link
    href="/release/2664677"
  >
    <a
      href="/release/2664677"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Caetano* E Chico* - Juntos E Ao Vivo
    </a>
  </Link>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FaTrash>
        <IconBase
          attr={
            Object {
              "viewBox": "0 0 448 512",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              key="0"
            />
          </svg>
        </IconBase>
      </FaTrash>
    </button>
  </Button>
  <br />
</FavoriteList>
`;
